src/pointer_constraints_unstable_v1.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright © 2020 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Alan Griffiths <alan@octopull.co.uk>
+ */
+
+#include "pointer_constraints_unstable_v1.h"
+
+#include "in_process_server.h"
+#include <version_specifier.h>
+
+wlcs::ZwpPointerConstraintsV1::ZwpPointerConstraintsV1(Client& client) :
+    manager{client.bind_if_supported<zwp_pointer_constraints_v1>(AnyVersion)}
+{
+}
+
+wlcs::ZwpPointerConstraintsV1::~ZwpPointerConstraintsV1() = default;
+
+wlcs::ZwpPointerConstraintsV1::operator zwp_pointer_constraints_v1*() const
+{
+    return manager;
+}
+
+wlcs::ZwpConfinedPointerV1::ZwpConfinedPointerV1(
+    ZwpPointerConstraintsV1& manager,
+    wl_surface* surface,
+    wl_pointer* pointer,
+    wl_region* region,
+    uint32_t lifetime) :
+    relative_pointer{zwp_pointer_constraints_v1_confine_pointer(manager, surface, pointer, region, lifetime)},
```

Comment:
Don't we generally format this like
```
    uint32_t lifetime)
    : relative_pointer...
```
?

Reply:
No, we normally format like this PR. What you show is [option 3](https://mir-server.io/doc/cppguide/?showone=Constructor_Initializer_Lists#Constructor_Initializer_Lists).

Reply:
Oh, I forgot we had multiple acceptable options there